cn0534/src/power_check.c

```diff
@@ -0,0 +1,1141 @@
+    fprintf(fd, buff);
+    fclose(fd);
+    if (state==0)
+    {
+        fd = fopen("/sys/class/leds/led0:green/delay_on", "w");
+        fprintf(fd, "50");
+        fclose(fd);
+        fd = fopen("/sys/class/leds/led0:green/delay_off", "w");
+        fprintf(fd, "50");
+        fclose(fd);
+    }
+#endif
+}
+
+static struct iio_context *scan(void)
+{
+    struct iio_scan_context *scan_ctx;
+    struct iio_context_info **info;
+    struct iio_context *ctx = NULL;
+    unsigned int i;
+    ssize_t ret;
+
+    scan_ctx = iio_create_scan_context(NULL, 0);
+    if (!scan_ctx)
+    {
+        fprintf(stderr, "Unable to create scan context\n");
+        return NULL;
+    }
+
+    ret = iio_scan_context_get_info_list(scan_ctx, &info);
+    if (ret < 0)
+    {
+        char err_str[1024];
+        iio_strerror(-ret, err_str, sizeof(err_str));
+        fprintf(stderr, "Scanning for IIO contexts failed: %s\n", err_str);
+        goto err_free_ctx;
+    }
+
+    if (ret == 0)
+    {
+        printf("No IIO context found.\n");
+        goto err_free_info_list;
+    }
+
+    if (ret == 1)
+    {
+        ctx = iio_create_context_from_uri(iio_context_info_get_uri(info[0]));
+    }
+    else
+    {
+        fprintf(stderr, "Multiple contexts found. Please select one using --uri:\n");
+
+        for (i = 0; i < (size_t)ret; i++)
+        {
+            fprintf(stderr, "\t%d: %s [%s]\n", i,
+                    iio_context_info_get_description(info[i]),
+                    iio_context_info_get_uri(info[i]));
+        }
+    }
+
+err_free_info_list:
+    iio_context_info_list_free(info);
+err_free_ctx:
+    iio_scan_context_destroy(scan_ctx);
+
+    return ctx;
+}
+
+struct write_l
+{
+    char *device;
+    char *channel;
+    bool in_out;
+    char *attribute;
+    char *value;
+};
+
+struct write_l write_log[128];
+static int wl_index = 0;
+
+static int32_t iio_set_attribute(char *device, char *channel, bool in_out,
+                              char *attribute, char *buffer, bool log)
+{
+    struct iio_device *dev;
+    struct iio_channel *chan;
+    const char *attr;
+    ssize_t ret;
+    char buf[1024];
+
+    dev = iio_context_find_device(ctx, device);
+    if (!dev)
+    {
+        fprintf(stderr, "Device %s not found\n", device);
+        iio_context_destroy(ctx);
+        return EXIT_FAILURE;
+    }
+
+    chan = iio_device_find_channel(dev, channel, in_out);
+    if (!chan)
+    {
+        fprintf(stderr, "%s channel not found in %s\n", channel, device);
+        iio_context_destroy(ctx);
+        return EXIT_FAILURE;
+    }
+
+    attr = iio_channel_find_attr(chan, attribute);
+    if (!attr)
+    {
+        fprintf(stderr, "%s attribute not found in %s:%s\n", attribute, device,
+                channel);
+        iio_context_destroy(ctx);
+        return EXIT_FAILURE;
+    }
+
+    ret = iio_channel_attr_read(chan, attribute, buf, sizeof(buf));
+    if (ret > 0)
+    {
+        /*if (log)
```

Comment:
is this a debug leftover ?

Reply:
unfortunately yes, removed this in 7969679